### PR TITLE
Updated Typescript examples for the multi-version API SDK

### PR DESCRIPTION
**Pull request by jkaster**
* also added a "download dashboard" example that supports rendering as PDF, PNG, or JPEG

### CODE REVIEW

**Comment by jkaster** [comment]
Just thought of some additional updates to make the "downloads" more flexible. You can wait for the next commit if you'd like.

**Comment by jkaster** [comment]
Ok, now it's ready for review

File: typescript/dual.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+const localConfig = '../looker.ini'
+const settings = new NodeSettingsIniFile(localConfig, "Looker")
+const session = new NodeSession(settings)
+const sdk = new Looker40SDK(session)
+const sdk31 = new Looker31SDK(session);
+
+(async () => {
+  const [me40, me31] = await Promise.all([sdk.ok(sdk.me()), sdk.ok(sdk31.me())])
+  if (me40.id == me31.id) {
+    console.log('Congratulations! You are using dual SDKs!')
```
Comment (rlm1023) [inline typescript/dual.ts]: 🎉

File: typescript/sudoAsUser.ts [hunk 2]
```diff
@@ -61,7 +61,8 @@ const matchDomain = '%@looker.com'
  * @returns {Promise<undefined | IUser>} Returns the first matched user, or undefined if no match
  */
 const anyoneButMe = async (userId: number, emailPattern: string) => {
-  const all = await sdk.ok(sdk.search_users({email: emailPattern, page: 0, per_page: 2}))
+  const all = await sdk.ok(sdk.search_users({email: emailPattern, page: 1, per_page: 2}))
+  // const all = await sdk.ok(sdk.search_users({email: emailPattern, page: 0, per_page: 2}))
```
Comment (rlm1023) [inline typescript/sudoAsUser.ts]: 🤔

File: typescript/utils.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  return dash
+}
+
+/**
+ * Get a tile by title from a dashboard
+ * @param {IDashboard} dash Dashboard to search
+ * @param {string} title Title title to find
+ * @returns {IDashboardElement | undefined} Returns the found tile or undefined
+ */
+export const getDashboardTile = (dash: IDashboard, title: string) => {
```
Comment (rlm1023) [inline typescript/utils.ts]: Interesting to do this look up by title. I don't think those are guaranteed unique.

File: typescript/downloadDashboard.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      return
+    }
+
+    const result = await waitForRender(sdk, task.id!)
+    fileName = `${dashboard.title}.${format}`
+    fs.writeFile(fileName, result, 'binary',(err) => {
+        if (err) {
+          fileName = undefined
+          console.error(err)}
+      }
```
[the remark this answers is not in the document]
Comment (jkaster) [reply typescript/downloadDashboard.ts]: Prettier is letting me down :(

File: typescript/downloadDashboard.ts [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    const fileName = await downloadDashboard(sdk, dashboard, outputFormat)
+    console.log(`open ${fileName} to see the download`)
+  }
+
+  await sdk.authSession.logout() // logout of API session
+  if (!sdk.authSession.isAuthenticated()) {
+    console.log('Logout successful')
+  }
+
+})()
```
[the remark this answers is not in the document]
Comment (jkaster) [reply typescript/downloadDashboard.ts]: Yeah, download the render while reporting progress with the simple callback

File: typescript/sudoAsUser.ts (hunk 2 above)
Comment (jkaster) [reply typescript/sudoAsUser.ts]: oh yuck. copy pasta. thanks for catching.

File: typescript/utils.ts (hunk 3 above)
Comment (jkaster) [reply typescript/utils.ts]: Nope, they're not. I make no guarantees with the sample code. Just intended to demonstrate SDK/API usage and downloading.

Comment (jkaster) [reply typescript/utils.ts]: You did inspire me to list the available dashboards by ID and title when the requested dashboard isn't found, though

**Pull request merged**